packages/console/src/pages/EnterpriseSsoDetails/Experience/DomainsInput/index.tsx

```diff
@@ -45,116 +35,28 @@ function DomainsInput({ className, values, onChange: rawOnChange, error, placeho
-              {option.value}
-              <IconButton
-                className={styles.delete}
-                size="small"
-                onClick={() => {
-                  handleDelete(option);
-                }}
-                onKeyDown={onKeyDownHandler(() => {
-                  handleDelete(option);
-                })}
-              >
-                <Close className={styles.close} />
-              </IconButton>
-            </Tag>
-          );
-        })}
-        <input
-          ref={inputRef}
-          // Need to use t() to complete the translation with prefix, use String() to convert the result to string.
-          // Should not show placeholder when there are values.
-          placeholder={conditional(values.length === 0 && placeholder && String(t(placeholder)))}
-          value={currentValue}
-          onKeyDown={(event) => {
-            if (event.key === 'Backspace' && currentValue === '') {
-              if (focusedValueId) {
-                onChange(values.filter(({ id }) => id !== focusedValueId));
-                setFocusedValueId(null);
-              } else {
-                setFocusedValueId(values.at(-1)?.id ?? null);
-              }
-              inputRef.current?.focus();
-            }
-            if (event.key === ' ' || event.code === 'Space' || event.key === 'Enter') {
-              // Focusing on input
-              if (currentValue !== '' && document.activeElement === inputRef.current) {
-                handleAdd(currentValue);
-              }
-              // Do not react to "Enter"
-              event.preventDefault();
-            }
-          }}
-          onChange={({ currentTarget: { value } }) => {
-            setCurrentValue(value);
-            setFocusedValueId(null);
-          }}
-          onFocus={() => {
-            inputRef.current?.focus();
-          }}
-          onBlur={() => {
-            if (currentValue !== '') {
-              handleAdd(currentValue);
-            }
-            setFocusedValueId(null);
-          }}
-        />
-      </div>
-      {Boolean(error) && typeof error === 'string' && (
-        <div className={styles.errorMessage}>{error}</div>
-      )}
-    </>
+    <MultiOptionInput
+      className={className}
+      values={values}
+      error={error}
+      placeholder={placeholder ? t(placeholder) : undefined}
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access
+      valueClassName={(option) => (option.status ? `tag ${option.status}` : undefined)}
+      renderValue={(option) => option.value}
+      validateInput={(text) => {
+        return {
+          value: {
+            value: text,
+            id: generateStandardShortId(),
+
+            // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-call
+            ...conditional(!domainRegExp.test(text) && { status: 'info' }),
+          },
+        };
+      }}
```

Comment:
Should provide a better validation handling logic here, with proper error messages instead.